.github/workflows/ci.yml

```diff
@@ -17,17 +17,25 @@ jobs:
 
     steps:
       - uses: actions/checkout@v4
+      - name: Install virtualenv
+        if: ${{ matrix.qgis-tags == 'latest' }}
+        # See PEP668
+        run: pip3 install -U --break-system-packages pip virtualenv
+      - name: Install virtualenv
+        if: ${{ matrix.qgis-tags != 'latest' }}
+        run: pip3 install -U pip virtualenv
       - name: Install dependencies
         run: |
-          python3 -m pip install -U pip setuptools
-          pip3 install -qr requirements.txt pytest-cov
-          pip3 install -e .
+          virtualenv --system-site-packages .venv
```

Comment:
This will break again if current latest is tagged with a release tag.
Is there a reason why you choose to use virtualenv package instead of native venv module?
Could we just install/update possibly missing venv with apt? Then there wouldn't be need for that special handling for different images.

```yml
      - name: Install virtualenv
        run: |
          apt update
          apt install -y python3-venv
      - name: Install dependencies
        run: |
          python3 -m venv --system-site-packages .venv
          .venv/bin/pip3 install -U pip setuptools
          .venv/bin/pip3 install -qr requirements.txt pytest-cov
          .venv/bin/pip3 install -e .
```